tests/structures/test_try_catch.py

```diff
@@ -912,3 +912,78 @@ def test_try_multiple_except_else_finally(self):
                 print("Do final cleanup")
             print('Done.')
             """, exits_early=True)
+
+    def test_except_an_expr(self):
+        self.assertCodeExecution("""
+            f = lambda: ValueError
+            try:
+               raise ValueError("YO")
+            except f():
+               print('Got a dynamic exception')
+            else:
+               print("Should do not print this")
+            finally:
+               print("Do final cleanup")
+            print('Done.')
+        """)
+
+        self.assertCodeExecution("""
+            f = lambda: ValueError
+            try:
+                raise ValueError
+            except f() as e:
+                print('Got a dynamic exception', e)
+            else:
+                print("Should do not print this")
+            finally:
+                print("Do final cleanup")
+            print('Done.')
+        """)
+
+        self.assertCodeExecution("""
+            f = lambda: ValueError
+            try:
+                raise TypeError("YO")
+            except f():
+                print("Should not print this")
+            else:
+                print("Do else handling")
+            finally:
+                print("Do final cleanup")
+            print('Done.')
+        """, exits_early=True)
+
+        self.assertCodeExecution("""
+            x = ValueError
+            try:
+                raise ValueError
+            except x as e:
+                print('Got a dynamic exception', e)
+            else:
+                print("Should do not print this")
+            finally:
+                print("Do final cleanup")
+            print('Done.')
+        """)
+
+    @expectedFailure
+    def test_except_tuple_custom_exception(self):
+        self.assertCodeExecution("""
+            class MyException(Exception):
+                pass
+            try:
+                raise MyException
+            except (MyException,):
+                print("Got a custom exception")
+            print('Done.')
+         """)
+
+        self.assertCodeExecution("""
```

Comment:
Can you please put this test in a separate method, so that it can run as well?

This seems to be a regression (i.e., if i'm not mistaken, this second `assertCodeExecution` passes on master)

Reply:
Done. Indeed, there seems to be a regression on `master`. But it seems it's the other way around: the second `assertCodeExecution` passes, while the first fails.

Do you think it's blocking the merge?

Reply:
Thanks!
Yeah, that's what I said -- the second passes. :)

So, while the patch seems to indeed already make things better, I would like to have a look with more patience to see what's causing that regression.
I'm a bit swamped lately but I'll have some days off soon and will look into it!